Find a functionally equivalent code:android.text.format.Time.normalize:COMMENT
<android.text.format.Time: long normalize(boolean)>
Method Modifier: public      native      
Method Modifier: Yes         Yes         
Method Updated:  Yes         No          
Comment:/**
 * Ensures the values in each field are in range. For example if the
 * current value of this calendar is March 32, normalize() will convert it
 * to April 1. It also fills in weekDay, yearDay, isDst and gmtoff.
 *
 * <p>
 * If "ignoreDst" is true, then this method sets the "isDst" field to -1
 * (the "unknown" value) before normalizing.  It then computes the
 * correct value for "isDst".
 *
 * <p>
 * See {@link #toMillis(boolean)} for more information about when to
 * use <tt>true</tt> or <tt>false</tt> for "ignoreDst".
 *
 * @return the UTC milliseconds since the epoch
 */

@@ -1 +1,6 @@
-
+{
+    calculator.copyFieldsFromTime(this);
+    long timeInMillis = calculator.toMillis(ignoreDst);
+    calculator.copyFieldsToTime(this);
+    return timeInMillis;
+}

